service/history/task/timer_standby_task_executor.go

```diff
@@ -81,11 +81,14 @@ func (t *timerStandbyTaskExecutor) Execute(
 	task Task,
 	shouldProcessTask bool,
 ) error {
-
-	timerTask, ok := task.GetInfo().(*persistence.TimerTaskInfo)
+	timer, ok := task.GetInfo().(persistence.Task)
 	if !ok {
 		return errUnexpectedTask
 	}
+	timerTask, err := timer.ToTimerTaskInfo()
+	if err != nil {
+		return err
```

Comment:
is this conversion error interpreted as a non-retriable error? (it should)

Reply:
Yeah, it's handled in the same way as errUnexpectedTask